src/display/text_layer.js

```diff
@@ -185,39 +205,42 @@ var renderTextLayer = (function renderTextLayerClosure() {
       }
 
       var width = ctx.measureText(textDiv.textContent).width;
-      textDiv.dataset.originalWidth = width;
+      textDivProperties.originalWidth = width;
       textLayerFrag.appendChild(textDiv);
 
-      var transform;
-      if (textDiv.dataset.canvasWidth !== undefined && width > 0) {
-        // Dataset values are of type string.
-        var textScale = textDiv.dataset.canvasWidth / width;
-        transform = 'scaleX(' + textScale + ')';
-      } else {
-        transform = '';
+      var transform = '';
+      if (textDivProperties.canvasWidth !== 0 && width > 0) {
+        var scale = textDivProperties.canvasWidth / width;
```

Comment:
I'm *really* sorry that I didn't think to ask about this before now, but I wonder if you couldn't store the `scale` in `textDivProperties.scale` (with `1` as the default value [above](https://github.com/mozilla/pdf.js/pull/7588/files#diff-986df9c532f6b9ab6cf81a23ce26e446R65)) instead?

That way you'd not need to either store `textDivProperties.originalWidth`,  nor recompute the `scale` in `expandTextDivs`. Instead, [below](https://github.com/mozilla/pdf.js/pull/7588/files#diff-986df9c532f6b9ab6cf81a23ce26e446R572) you'd just do e.g.
```javascript
...
var scale = divProperties.scale;

if (scale !== 1) {
  transform = 'scaleX(' + scale + ')';
}
...
```